### PR TITLE
[Kettle] Skip stream of jobs that are excluded

### PR DESCRIPTION
**Pull request by MushuEE**
This is not the most pretty but there was no clean way I could think of without changing a whole lot more. I also added tests that give a better picture of how everything is handled

#21193

/area kettle

### CODE REVIEW

File: kettle/stream.py [hunk 1]
```diff
@@ -41,23 +42,30 @@
 import make_db
 import make_json
 
+
 MAX_ROW_UPLOAD = 10 # See https://github.com/googleapis/google-cloud-go/issues/2855
 
-def process_changes(results):
+
+def should_exclude(objectID, bucketID, buckets):
+    job_name = objectID.rsplit('/')[-3] # See tests for refernce
```
Comment (amwat) [inline kettle/stream.py:50]: i think a comment of a sample objectID would also go well here [-3] looks like magic :)

Comment (MushuEE) [inline kettle/stream.py:50]:
```suggestion
   # Objects of form a/b/c/<jobname>/<hash>/<objectFile>'
    job_name = objectID.rsplit('/')[-3] # See tests for refernce
```

Comment (MushuEE) [inline kettle/stream.py:51]:
```suggestion
    job_name = objectID.rsplit('/')[-3] # See tests for reference
```

Comment (MushuEE) [reply kettle/stream.py:51]: done

**Pull request merged**